### PR TITLE
Add timezones to non-test logging

### PR DESCRIPTION
**Pull request by andr3ej**
* Consistently use comma as millis separator
* Move test config file so it's available in intellij without gradle assemble
* Explicit logging config for nodes, so the test config is overridden when test-utils is in use
* Remove redundant node dependencies directory and classpath entries, this fixes the multiple slf4j bindings warning
* Unduplicate caplet code
* Make 2 tests runnable in any working dir

### CODE REVIEW

File: verifier/build.gradle [hunk 1]
```diff
@@ -23,7 +23,7 @@ sourceSets {
 // Use manual resource copying of log4j2.xml rather than source sets.
 // This prevents problems in IntelliJ with regard to duplicate source roots.
 processResources {
-    from file("$rootDir/config/dev/log4j2.xml")
+    from file("$rootDir/config/dev/log4j2-corda.xml")
```
Comment (chrisr3) [inline verifier/build.gradle]: Extra work.

File: webserver/build.gradle [hunk 2]
```diff
@@ -20,7 +20,7 @@ sourceSets {
 }
 
 processResources {
-    from file("$rootDir/config/dev/log4j2.xml")
+    from file("$rootDir/config/dev/log4j2-corda.xml")
```
Comment (chrisr3) [inline webserver/build.gradle]: Extra work

File: webserver/webcapsule/build.gradle [hunk 3]
```diff
@@ -25,16 +25,17 @@ task buildWebserverJar(type: FatCapsule, dependsOn: project(':node').compileJava
     applicationSource = files(
             project(':webserver').configurations.runtime,
             project(':webserver').jar,
+            new File(project(':node').buildDir, 'classes/main/AbstractCordaCaplet.class'),
             new File(project(':node').buildDir, 'classes/main/CordaCaplet.class'),
-            new File(project(':node').buildDir, 'classes/main/CordaCaplet$1.class'),
-            "$rootDir/config/dev/log4j2.xml"
+            "$rootDir/config/dev/log4j2-corda.xml"
```
Comment (chrisr3) [inline webserver/webcapsule/build.gradle]: Extra work.

File: webserver/webcapsule/build.gradle [hunk 4]
```diff
@@ -25,16 +25,17 @@ task buildWebserverJar(type: FatCapsule, dependsOn: project(':node').compileJava
...
     )
     from 'NOTICE' // Copy CDDL notice
 
     capsuleManifest {
         applicationVersion = corda_release_version
         javaAgents = ["quasar-core-${quasar_version}-jdk8.jar"]
         systemProperties['visualvm.display.name'] = 'Corda Webserver'
+        systemProperties['log4j.configurationFile'] = 'classpath:log4j2-corda.xml'
```
Comment (chrisr3) [inline webserver/webcapsule/build.gradle]: Extra work.

File: client/rpc/build.gradle [hunk 5]
```diff
@@ -36,9 +36,6 @@ sourceSets {
 }
 
 processSmokeTestResources {
-    from(file("$rootDir/config/test/log4j2.xml")) {
-        rename 'log4j2\\.xml', 'log4j2-test.xml'
-    }
     from(project(':node:capsule').tasks.buildCordaJAR) {
```
Comment (chrisr3) [inline client/rpc/build.gradle]: Again, why remove `log4j2-test.xml` from the smoke test resources?

File: node/capsule/build.gradle [hunk 6]
```diff
@@ -25,9 +25,9 @@ task buildCordaJAR(type: FatCapsule, dependsOn: project(':node').compileJava) {
     applicationSource = files(
             project(':node').configurations.runtime,
             project(':node').jar,
+            '../build/classes/main/AbstractCordaCaplet.class',
             '../build/classes/main/CordaCaplet.class',
-            '../build/classes/main/CordaCaplet$1.class',
-            "$rootDir/config/dev/log4j2.xml"
+            "$rootDir/config/dev/log4j2-corda.xml"
```
Comment (chrisr3) [inline node/capsule/build.gradle]: Extra work...

File: test-utils/build.gradle [hunk 7]
```diff
@@ -27,6 +27,7 @@ sourceSets {
 }
 
 dependencies {
+    compile project(':smoke-test-utils')
```
Comment (chrisr3) [inline test-utils/build.gradle]: Why does `test-utils` depend upon `smoke-test-utils` now? If it's only to import `log42-test.xml` into other projects "by the back door" then I strongly disagree. We had a `processTestResources` task to do that.

File: smoke-test-utils/src/main/resources/log4j2-test.xml [hunk 8]
```diff
@@ -5,7 +5,7 @@
     </Properties>
```
Comment (chrisr3) [inline smoke-test-utils/src/main/resources/log4j2-test.xml]: I don't think this file belongs in `smoke-test-utils` like this, even if it probably did need renaming to `log4j2-test.xml`.

File: test-utils/src/main/kotlin/net/corda/testing/driver/ProcessUtilities.kt [hunk 9]
```diff
@@ -26,20 +26,18 @@ object ProcessUtilities {
...
-            listOf("-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=$jdwpPort")
-        } else {
-            emptyList()
+        val allArguments = mutableListOf<String>().apply {
+            add((System.getProperty("java.home") / "bin" / "java").toString())
+            if (jdwpPort != null) add("-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=$jdwpPort")
+            add("-Xmx200m")
+            add("-XX:+UseG1GC")
+            addAll(extraJvmArguments)
+            add("-Dlog4j.configurationFile=classpath:log4j2-corda.xml")
```
Comment (chrisr3) [inline test-utils/src/main/kotlin/net/corda/testing/driver/ProcessUtilities.kt]: Extra work.

File: tools/explorer/capsule/build.gradle [hunk 10]
```diff
@@ -31,13 +31,15 @@ task buildExplorerJAR(type: FatCapsule, dependsOn: project(':tools:explorer').co
...
         project(':tools:explorer').jar,
+        new File(project(':node').buildDir, 'classes/main/AbstractCordaCaplet.class'),
         '../build/classes/main/ExplorerCaplet.class'
     )
     classifier 'fat'
 
     capsuleManifest {
         applicationVersion = corda_release_version
         systemProperties['visualvm.display.name'] = 'Node Explorer'
+        systemProperties['log4j.configurationFile'] = 'classpath:log4j2-corda.xml'
```
Comment (chrisr3) [inline tools/explorer/capsule/build.gradle]: Extra work.

File: tools/demobench/src/test/resources/log4j2-test.xml [hunk 11]
```diff
@@ -3,7 +3,7 @@
 
     <Appenders>
         <Console name="Console-Appender" target="SYSTEM_OUT">
-            <PatternLayout pattern="%date %highlight{%level %c{1}.%M - %msg%n}{INFO=white,WARN=red,FATAL=bright red}"/>
+            <PatternLayout pattern="%d %highlight{%level %c{1}.%M - %msg%n}{INFO=white,WARN=red,FATAL=bright red}"/>
```
Comment (andr3ej) [inline tools/demobench/src/test/resources/log4j2-test.xml]: Consistency, we use %d everywhere else

File: build.gradle [hunk 12]
```diff
@@ -103,12 +103,6 @@ allprojects {
     sourceCompatibility = 1.8
     targetCompatibility = 1.8
 
-    // Use manual resource copying of log4j2.xml rather than source sets.
-    // This prevents problems in IntelliJ with regard to duplicate source roots.
-    processTestResources {
-        from file("$rootDir/config/test/log4j2.xml")
-    }
-
```
Comment (andr3ej) [inline build.gradle]: Primary reason is to have logging in IntelliJ before gradle assemble / after gradle clean. Also I don't understand why we're copying it around manually if the usual resources mechanism can do it. It would make sense if we're filtering the file, but we're not

File: config/dev/log4j2-corda.xml [hunk 13]
```diff
@@ -13,7 +13,7 @@
```
Comment (andr3ej) [inline config/dev/log4j2-corda.xml]: When test-utils is in the classpath, log4j2-test.xml needs to be overridden. And if I override it with log4j2.xml, someone is likely to remove that because it's the default

File: node/src/main/java/AbstractCordaCaplet.java [hunk 14]
```diff
@@ -0,0 +1,29 @@
+import java.io.File;
```
Comment (andr3ej) [inline node/src/main/java/AbstractCordaCaplet.java]: It does - only CordaCaplet (in node) and ExplorerCaplet (depends on node) use it

File: smoke-test-utils/src/main/resources/log4j2-test.xml (hunk 8 above)
Comment (andr3ej) [inline smoke-test-utils/src/main/resources/log4j2-test.xml]: The bits of gradle that this replaces were logically doing the exact same thing, in a more obfuscated way

File: smoke-test-utils/src/main/resources/log4j2-test.xml [hunk 15]
```diff
@@ -5,7 +5,7 @@
...
     <Appenders>
         <Console name="Console-Appender" target="SYSTEM_OUT">
-            <PatternLayout pattern="[%-5level] %d{HH:mm:ss.SSS} [%t] %c{2}.%M - %msg%n" />
+            <PatternLayout pattern="[%-5level] %d{HH:mm:ss,SSS} [%t] %c{2}.%M - %msg%n" />
```
Comment (andr3ej) [inline smoke-test-utils/src/main/resources/log4j2-test.xml]: We use comma everywhere else via ISO8601 default

File: test-utils/build.gradle (hunk 7 above)
Comment (andr3ej) [inline test-utils/build.gradle]: No, we had 2 processTestResources tasks to do that

File: node/build.gradle [hunk 16]
```diff
@@ -55,7 +55,7 @@ sourceSets {
 // Use manual resource copying of log4j2.xml rather than source sets.
 // This prevents problems in IntelliJ with regard to duplicate source roots.
 processResources {
-    from file("$rootDir/config/dev/log4j2.xml")
+    from file("$rootDir/config/dev/log4j2-corda.xml")
```
Comment (andr3ej) [inline node/build.gradle]: It's exactly the same amount of code

File: node/capsule/build.gradle (hunk 6 above)
Comment (andr3ej) [inline node/capsule/build.gradle]: Actually less code

File: node/capsule/build.gradle [hunk 17]
```diff
@@ -40,6 +40,7 @@ task buildCordaJAR(type: FatCapsule, dependsOn: project(':node').compileJava) {
         // javaAgents = ["quasar-core-${quasar_version}-jdk8.jar=${quasarExcludeExpression}"]
         javaAgents = ["quasar-core-${quasar_version}-jdk8.jar"]
         systemProperties['visualvm.display.name'] = 'Corda'
+        systemProperties['log4j.configurationFile'] = 'classpath:log4j2-corda.xml'
```
Comment (andr3ej) [inline node/capsule/build.gradle]: This is only extra work * n because the capsule config is duplicated so many times

File: tools/explorer/build.gradle [hunk 18]
```diff
@@ -15,7 +15,7 @@ mainClassName = 'net.corda.explorer.Main'
 // Use manual resource copying of log4j2.xml rather than source sets.
 // This prevents problems in IntelliJ with regard to duplicate source roots.
 processResources {
-    from file("$rootDir/config/dev/log4j2.xml")
+    from file("$rootDir/config/dev/log4j2-corda.xml")
 }
```
Comment (andr3ej) [inline tools/explorer/build.gradle]: Nope, same amount of code

File: verifier/build.gradle (hunk 1 above)
Comment (andr3ej) [inline verifier/build.gradle]: Identical complexity

File: webserver/build.gradle (hunk 2 above)
Comment (andr3ej) [inline webserver/build.gradle]: This is a rename

File: webserver/webcapsule/build.gradle (hunk 3 above)
Comment (andr3ej) [inline webserver/webcapsule/build.gradle]: 2 red lines, 1 green

File: webserver/webcapsule/build.gradle (hunk 4 above)
Comment (andr3ej) [inline webserver/webcapsule/build.gradle]: Capsule config duplication can be fixed in another PR

File: client/rpc/build.gradle (hunk 5 above)
Comment (andr3ej) [inline client/rpc/build.gradle]: It's in there but we no longer have to copy manually

File: test-utils/src/main/kotlin/net/corda/testing/driver/ProcessUtilities.kt (hunk 9 above)
Comment (andr3ej) [inline test-utils/src/main/kotlin/net/corda/testing/driver/ProcessUtilities.kt]: Because similar chunks of code are everywhere already. Making this configurable in one place can go in a separate PR

File: samples/network-visualiser/build.gradle [hunk 19]
```diff
@@ -41,6 +41,7 @@ task deployVisualiser(type: FatCapsule) {
     applicationClass 'net.corda.netmap.NetworkMapVisualiser'
     reallyExecutable
     capsuleManifest {
+        systemProperties['log4j.configurationFile'] = 'classpath:log4j2-corda.xml'
```
Comment (andr3ej) [inline samples/network-visualiser/build.gradle]: It took me a while to figure out whether we rely on automagic config, or maybe it is explicit but subtle? This way is clearest

**Comment by andr3ej** [comment]
I've spun off refactors in the past to very inconsistent reviews, so I'm not keen to split this one up, especially as the headline ticket is so trivial. The logging experience enhancements belong here anyway as they are helpful when testing the timezone changes. The answers to most inline comments were already in my top comment

File: build.gradle (hunk 12 above)
Comment (mikehearn) [inline build.gradle]: Ask Matthew about this. The prior way we did it broke IntelliJ.

File: node/capsule/build.gradle (hunk 17 above)
Comment (chrisr3) [inline node/capsule/build.gradle]: I was referring to having to set `log4j.configurationFile` now, because we didn't need to do this before. And I don't believe we should have to do so now either.

File: tools/demobench/src/test/resources/log4j2-test.xml (hunk 11 above)
Comment (chrisr3) [inline tools/demobench/src/test/resources/log4j2-test.xml]: Big deal. %date is _still_ more explicit.

File: config/dev/log4j2-corda.xml (hunk 13 above)
Comment (chrisr3) [inline config/dev/log4j2-corda.xml]: I suggest that including `test-utils` in the classpath should _not_ affect how the logging behaves in the first place. And this kind of side-effect is probably why this PR should just be concentrating on changing the timezone formatting.

File: node/build.gradle (hunk 16 above)
Comment (chrisr3) [inline node/build.gradle]: I meant that it's an extra change that only exists because `log4j2.xml` has been renamed.

File: node/src/main/java/AbstractCordaCaplet.java (hunk 14 above)
Comment (chrisr3) [inline node/src/main/java/AbstractCordaCaplet.java]: Except that Node Explorer is actually an RPC client that _shouldn't_ depend on node.

File: smoke-test-utils/src/main/resources/log4j2-test.xml (hunk 8 above)
Comment (chrisr3) [inline smoke-test-utils/src/main/resources/log4j2-test.xml]: `smoke-test-utils` is a brand new module full of code that I wrote and there is simply no way that `test-utils` can suddenly need to depend upon it,

File: tools/explorer/build.gradle (hunk 18 above)
Comment (chrisr3) [inline tools/explorer/build.gradle]: Another diff that shouldn't need to exist.

File: verifier/build.gradle (hunk 1 above)
Comment (chrisr3) [inline verifier/build.gradle]: Shouldn't need to exist.

File: webserver/build.gradle (hunk 2 above)
Comment (chrisr3) [inline webserver/build.gradle]: A _bad_ rename.

File: test-utils/build.gradle (hunk 7 above)
Comment (chrisr3) [inline test-utils/build.gradle]: I think we actually had 1 `processTestResources` and 1 `processSmokeTestResources`. and that wasn't a problem that needed fixing. It certainly has nothing to do with adding timezones to non-test logging.

File: tools/explorer/capsule/build.gradle (hunk 10 above)
Comment (chrisr3) [inline tools/explorer/capsule/build.gradle]: Extra work = "You only needed to add this line because you renamed `log4j2.xml` to `log4j2-corda.xml`".

File: smoke-test-utils/src/main/resources/log4j2-test.xml (hunk 15 above)
Comment (chrisr3) [inline smoke-test-utils/src/main/resources/log4j2-test.xml]: The British decimal separator is _still_ the full-stop and not the comma, regardless of whatever default value Log4J2 may be configured with.

File: build.gradle (hunk 12 above)
Comment (chrisr3) [inline build.gradle]: You're asking that _every_ test project now depend on `smoke-test-utils` _just_ to obtain a copy of `log4j2-test.xml` on the classpath. This is madness. There is _no_ logical connection between "smoke tests" and "logging configuration".
